src/com/dabomstew/pkrandom/romhandlers/Gen3RomHandler.java

```diff
@@ -1386,12 +1401,19 @@ public boolean hasWildAltFormes() {
             tr.poketype = pokeDataType;
             tr.name = this.readVariableLengthString(trOffset + 4);
             tr.fullDisplayName = tcnames.get(trainerclass) + " " + tr.name;
-            // Pokemon data!
+            // Pokemon structure data is like
+            // IV IV LV SP SP
+            // (HI HI)
+            // (M1 M1 M2 M2 M3 M3 M4 M4)
+            // IV is a "difficulty" level between 0 and 255 to represent 0 to 31 IVs.
+            //     These IVs affect all attributes. For the vanilla games, the majority
+            //     of trainers have 0 IVs; Elite Four members will have 31 IVs.
+            // https://github.com/pret/pokeemerald/blob/6c38837b266c0dd36ccdd04559199282daa7a8a0/include/data.h#L22
             if (pokeDataType == 0) {
                 // blocks of 8 bytes
                 for (int poke = 0; poke < numPokes; poke++) {
                     TrainerPokemon thisPoke = new TrainerPokemon();
-                    thisPoke.AILevel = readWord(pointerToPokes + poke * 8);
+                    thisPoke.IVs = ((readWord(pointerToPokes + poke * 8) & 0xFF) * 31) / 255;

```

Comment:
I have a question about the division that's being done when reading/writing for Gen 3-5. We're dealing with `int`s here, so it's integer division - wouldn't this result in the written value being different than the one that's read? As an example, a Pokemon with "difficulty level" 250:
```
(250 * 31) / 255 = 30
(30 * 255) / 31 = 246
```

Reply:
As far as I can tell, the game also does integer division when it computes IVs, so 246 difficulty is the same as 250.

Reply:
246 wouldn't give the same IVs as 250:

```
(250 * 31) / 255 = 30
(246 * 31) / 255 = 29
```

Reply:
247 does, so if you don't want to store this in the `strength` byte, you can do float division and then just round up. Or you can just use the `strength` byte since we don't modify the IVs.

Reply:
I feel a little embarrassed to have overlooked this. My original attempt was off for everything except IVs = 0 and IVs = 31 🙃 
I made sure my spreadsheet was flooring, not rounding this time and assured myself that always rounding up when calculating difficulty (capping at 255), will come out in the game correctly.

I prefer to use the IVs to calculate the strength because if we *do* want to randomize/modify the IVs, then we have less work to make that happen.

Reply:
Please take a look at the modified setTrainers for Gens 3, 4, and 5 where we compute the strength/difficulty based on IVs. Gen6 passed the byte already (ORAS/XY split), so it shouldn't need a change.